### PR TITLE
Enumerables_my_module:

### PR DESCRIPTION
**Pull request by SamanAtashi**
# Enumerables_my_module:

1. my_each
2. my_each_with_index
3. my_select
4. my_all?
5. my_any?
6. my_none?
7. my_count
8. my_map
9. my_inject
+ multiply_els

### CODE REVIEW

**Comment by mimipeshy** [comment]
## STATUS : CHANGES REQUESTED 🔴 ♻️ 

Hello team @SamanAtashi and @danmainah 👋🏾 👋🏾 , this is a strong first submission, great job! 😄 

#### Highlights

- Linter checks are passing ✔️ 
- Your README fie is well written ✔️ 
- A good number of your tests are passing ✔️

#### To Improve

- [ ] Check out inline comments on what can be improved
- [x] In your README file, add instructions on how to run your project

That's all. 😄 , kindy submit another review when you are done

Regards
Peris Ndanu
_If you have any questions, kindly leave a comment here and tag my name @mimipeshy and I will be glad to help_

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+          yield self[i]
+        else
+          yield(keys[i], values[i])
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
```
Comment (mimipeshy) [inline enumerables.rb]: - [x]  enumerables #my_each returns an Enumerator if no block is given
- [x]  enumerables #my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
    ` Failure/Error: expect(array.my_each(&block)).to eq(array.each(&block)) `
- [x]  enumerables #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
    ``` NameError:
       undefined local variable or method `length' for 5..50:Range
   ```
- [x] enumerables #my_each when a block is given when self is a range behaves like #each calls the given block once for each element in self
    ```  NameError:  undefined local variable or method `length' for 5..50:Range ```
- [x]  enumerables #my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self
  ```  expected: "[:a, 1][:b, 2][:c, 3][:d, 4][:e, 5]" but  got: "abcde" ```

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        else
+          yield(values[i], i)
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
```
Comment (mimipeshy) [inline enumerables.rb]: - [x] enumerables #my_each_with_index returns an enumerator if no block is given
- [x]  enumerables #my_each_with_index when a block is given when self is an array returns the array itself after calling the given block once for each element in self
- [x] enumerables #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self
``` NameError: undefined local variable or method `length' for 5..50:Range ```
- [x] enumerables #my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
``` NameError: undefined local variable or method `length' for 5..50:Range ```
- [x]  enumerables #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self
```
expected: "Num: [:a, 1], idx: 0\nNum: [:b, 2], idx: 1\nNum: [:c, 3], idx: 2\nNum: [:d, 4], idx: 3\nNum: [:e, 5], idx: 4\n"
got: "Num: 1, idx: 0\nNum: 2, idx: 1\nNum: 3, idx: 2\nNum: 4, idx: 3\nNum: 5, idx: 4\n"

```

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      end
+
+      return new_arr if instance_of?(Array)
+      return new_hash if instance_of?(Hash)
+
+    else
+      puts 'block was not given'
+    end
+  end
+
```
Comment (mimipeshy) [inline enumerables.rb]: - [x]  enumerables #my_select returns an enumerator if no block is given
- [x] enumerables #my_select returns an array containing all elements of enum for which the given block returns a true value::range
``` NameError: undefined local variable or method `length' for 5..50:Range ```

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        elsif sth.is_a? Class
+          return false unless item.is_a? sth
+        elsif [false, nil].include?(item)
+          return false
+        end
+      end
+    end
+    true
+  end
+
```
Comment (mimipeshy) [inline enumerables.rb]: - [x]  enumerables #my_all? returns false if the block returns false or nil::range
      ``` Failure/Error: expect(range.my_all?(&false_block)).to eq(range.all?(&false_block)) ```      
      ``` NoMethodError: undefined method `empty?' for 5..50:Range ```

- [x]  enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
      ``` Failure/Error: expect(array.my_all?(3)).to be array.all?(3) ```
      ``` expected false but got true ```

File: enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+          return true if item.is_a? sth
+        else
+          return false if self == [nil] || self == [false] || self == [nil, false] || self == [false, nil]
+
+          return true
+        end
+      end
+    end
+    false
+  end
```
Comment (mimipeshy) [inline enumerables.rb]: - [x] enumerables #my_any? returns false if the block ever returns a value false or nil::range
      Failure/Error: expect(range.my_any?(&false_block)).to eq(range.any?(&false_block))
      
      NoMethodError:
        undefined method `empty?' for 5..50:Range

- [x] enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
      Failure/Error: expect(false_array.my_any?).to be false_array.any?
      
        expected false
             got true

- [x] enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
      
        expected false
             got true

File: enumerables.rb [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+          return false if item.is_a? sth
+        else
+          return true if self == [nil] || self == [false] || self == [nil, false] || self == [false, nil]
+
+          return false
+        end
+      end
+    end
+    true
+  end
```
Comment (mimipeshy) [inline enumerables.rb]: - [x] enumerables #my_none? returns false if the block ever returns true for all elements::range
      Failure/Error: expect(range.my_none?(&false_block)).to eq(range.none?(&false_block))
      
      NoMethodError:
        undefined method `empty?' for 5..50:Range

- [x] enumerables #my_none? when no block or argument is given returns true only if none of the collection members is true
      Failure/Error: expect(false_array.my_none?).to be true
      
        expected true
             got false

- [x] enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
        expected true
             got false

File: enumerables.rb [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+      to_a.my_each do |item|
+        result.push(proc.call(item))
+      end
+    else
+      to_a.my_each do |item|
+        result.push(yield item)
+      end
+    end
+    result
+  end
```
Comment (mimipeshy) [inline enumerables.rb]: - [x] enumerables #my_map returns an Enumerator if no block is given
      Failure/Error: expect(array.my_map).to be_an(Enumerator)

- [x] enumerables #my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
      Failure/Error: expect(array.my_map(my_proc) { |num| num < 10 }).to eq array.map(&my_proc)

File: enumerables.rb [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+      now_it_is = sth.length > 1 ? sth.shift : new_arr.shift
+      new_arr.my_each do |item|
+        now_it_is = now_it_is.send(sth[0].to_s, item)
+      end
+      return now_it_is
+    end
+    now_it_is
+  end
+end
+
```
Comment (mimipeshy) [inline enumerables.rb]: - [x] enumerables #my_inject raises a "LocalJumpError" when no block or argument is given
      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)

- [x]  enumerables #my_inject does not mutate the original array

- [x] enumerables #my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: expect(array.my_inject(&operation)).to eq array.inject(&operation)

- [x]  enumerables #my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range
      Failure/Error: actual = range.my_inject { |prod, n| prod * n }
      
      NoMethodError:
        undefined method `shift' for 5..50:Range

- [x] enumerables #my_inject when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }
      
      NameError:
        undefined local variable or method `length' for 5..50:Range

- [x]  enumerables #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
      Failure/Error: expect(range.my_inject(:*)).to eq range.inject(:*)
      
      NoMethodError:
        undefined method `shift' for 5..50:Range

- [x]  enumerables #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range
      Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)
      
      NameError:
        undefined local variable or method `length' for 5..50:Range

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @SamanAtashi @danmainah 
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

There are 16 test cases that are failing. Kindly work on the suggested changes  👇.
### my_each
- [x] #my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
- [x] #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
### my_each_with_index
- [x] #my_each_with_index when a block is given when self is an array returns the array itself after calling the given block once for each element in self
- [x] #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self
### my_all
- [x] #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
### my_any
- [x] #my_any? when no block or argument is given returns false if at least one of the collection is not true
- [x] #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
### my_none
- [x] #my_none? when no block or argument is given returns true only if none of the collection members is true
- [x] #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
### my_map
- [x] #my_map returns a new array with the results of running a given block once for every element in enum.
- [x] #my_map returns a new array with the results of running a given block once for every element in enum.
- [x] #my_map returns a new array with the results of running a given block once for every element in enum::range
- [x] #my_map accepts a proc as an argument and returns a new array with the results of running the proc once for every element in enum.
- [x] #my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
- [x] #my_map does not mutate the original array
### my_inject
- [x] #my_inject raises a "LocalJumpError" when no block or argument is given
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad (Code Reviewer)](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Comment by SamanAtashi** [comment]
@umairarshadbutt

Hello, thanks for the review. 
Can I ask for the tests or some sample because we can not find any good examples for these tests to check them?!

Thanks.

**Comment by umairarshadbutt** [comment]
Hi @SamanAtashi 
Sure, I will share more details within 30 minutes

Best,
Umair Arshad

**Comment by umairarshadbutt** [comment]
###  Hi @SamanAtashi These are the variables I used in the tests below: :point_down:
```
ARRAY_SIZE = 100
LOWEST_VALUE = 0
HIGHEST_VALUE = 9

describe 'enumerables' do
  let(:array) { Array.new(ARRAY_SIZE) { rand(LOWEST_VALUE...HIGHEST_VALUE) } }
  let(:block) { proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 } }
  let(:words) { %w[dog door rod blade] }
  let(:range) { Range.new(5, 50) }
  let(:hash) { { a: 1, b: 2, c: 3, d: 4, e: 5 } }
  let(:numbers) { [1, 2i, 3.14] }
  let!(:array_clone) { array.clone }

```
### my_each
- #my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
     Failure/Error: expect(array.my_each(&block)).to eq(array.each(&block))
     
       expected: [5, 3, 3, 5, 6, 2, 6, 6, 4, 4, 3, 3, 3, 2, 3, 4, 3, 2, 7, 6, 0, 2, 3, 5, 0, 0, 7, 5, 6, 0, 7, 7, 2, 7..., 3, 2, 7, 0, 3, 1, 3, 1, 6, 3, 2, 5, 3, 5, 0, 4, 8, 5, 3, 4, 6, 6, 4, 5, 6, 8, 6, 3, 6, 6, 3, 6, 3]
            got: nil
     
- #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
     Failure/Error: expect(range.my_each(&block)).to eq(range.each(&block))
     
       expected: 5..50
            got: nil
     
### my_each_with_index
- #my_each_with_index when a block is given when self is an array returns the array itself after calling the given block once for each element in self
     Failure/Error: expect(array.my_each_with_index(&block)).to eq(array.each_with_index(&block))
     
       expected: [1, 1, 0, 7, 6, 6, 5, 6, 8, 8, 0, 7, 2, 7, 8, 8, 1, 2, 2, 1, 5, 4, 0, 6, 5, 3, 0, 2, 1, 5, 6, 6, 4, 0..., 5, 5, 6, 4, 6, 4, 2, 0, 2, 1, 7, 5, 3, 4, 2, 3, 7, 5, 1, 8, 5, 6, 8, 8, 2, 1, 5, 4, 0, 7, 3, 8, 6]
            got: nil

- #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self
     Failure/Error: expect(range.my_each_with_index(&block)).to eq(range.each_with_index(&block))
     
       expected: 5..50
            got: nil
     
### my_all

- #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
     Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
     
       expected false
            got true
  
### my_any
- #my_any? when no block or argument is given returns false if at least one of the collection is not true
     Failure/Error: expect(false_array.my_any?).to be false_array.any?
     
       expected false
            got true
    
- #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
     Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
       expected false
            got true

### my_none
- #my_none? when no block or argument is given returns true only if none of the collection members is true
     Failure/Error: expect(false_array.my_none?).to be true
     
       expected true
            got false
- #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
     Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
     
       expected true
            got false

### my_map
- #my_map returns a new array with the results of running a given block once for every element in enum.
      Failure/Error: expect(array.my_map { |num| num < 10 }).to eq(array.map { |num| num < 10 })
      
      ArgumentError:
        tried to create Proc object without a block

- #my_map returns a new array with the results of running a given block once for every element in enum.
      Failure/Error: expect(array.my_map(&block)).to eq array.map(&block)
      
      ArgumentError:
        tried to create Proc object without a block
- #my_map returns a new array with the results of running a given block once for every element in enum::range
      Failure/Error: expect(range.my_map(&block)).to eq range.map(&block)
      
      ArgumentError:
        tried to create Proc object without a block

- #my_map accepts a proc as an argument and returns a new array with the results of running the proc once for every element in enum.
      Failure/Error: expect(array.my_map(&my_proc)).to eq array.map(&my_proc)
      
      ArgumentError:
        tried to create Proc object without a block

- #my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
      Failure/Error: expect(array.my_map(my_proc) { |num| num < 10 }).to eq array.map(&my_proc)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- #my_map does not mutate the original array
      Failure/Error: array.my_map { |num| num + 1 }
      
      ArgumentError:
        tried to create Proc object without a block

### my_inject
- #my_inject raises a "LocalJumpError" when no block or argument is given
      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
      
        expected LocalJumpError, 
        got #<NoMethodError: undefined method `' for 5:Integer> with backtrace:

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @SamanAtashi @danmainah
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

There are just 3 test cases that are failing. Kindly work on the suggested changes point_down.
- [x] #my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [x] #my_any? when no block or argument is given returns false if at least one of the collection is not true
- [x] #my_none? when no block or argument is given returns true only if none of the collection members is true
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad (Code Reviewer)](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @SamanAtashi and @danmainah :wave: 
You've done well by making the changes required by the previous TSE 👏 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

* [x] #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
* [x] #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
* [x] #my_none? when no block or argument is given returns true only if none of the collection members is true
* [x] #my_none? when no block or argument is given returns false only if one of the collection members is true
* [x] #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
* [x] #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern




## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Comment by SamanAtashi** [comment]
@ElijahTobs
Hello sir, Thanks for the review.
Can I ask for some help or a small hint ?! 
We are really confused because our tests pass with correct answer but when we ask for review we have problem.

**Comment by ElijahTobs** [comment]
Alright. Give me about 20 minutes

**Comment by ElijahTobs** [comment]
## Variables used in the test
```ruby
ARRAY_SIZE = 100
LOWEST_VALUE = 0
HIGHEST_VALUE = 9

describe 'enumerables' do
  let(:array) { Array.new(ARRAY_SIZE) { rand(LOWEST_VALUE...HIGHEST_VALUE) } }
  let(:block) { proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 } }
  let(:words) { %w[dog door rod blade] }
  let(:range) { Range.new(5, 50) }
  let(:hash) { { a: 1, b: 2, c: 3, d: 4, e: 5 } }
  let(:numbers) { [1, 2i, 3.14] }
  let!(:array_clone) { array.clone }
```



### my_all?
- when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
     _Failure/Error: expect(array.my_all?(3)).to be array.all?(3)_
     
```ruby
expected false
     got true
```

### my_any?
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
     _Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')_
```ruby
       expected false
            got true
```
### my_none?
- when no block or argument is given returns true only if none of the collection members is true
     _Failure/Error: expect(false_array.my_none?).to be true_
 ```ruby 
       expected true
            got #<String:47268029661260> => "last false"
```
- when no block or argument is given returns false only if one of the collection members is true
     _Failure/Error: expect(true_array.my_none?).to be false_
```ruby
     expected false
            got #<String:47268029714260> => "last false"
```
- when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
     Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
```ruby  
       expected true
            got false
```
- when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
     Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
```ruby
       expected false
            got #<String:47268029948960> => "last false"
```

**Review by jubaan — CHANGES_REQUESTED** [review]
[![Microverse][microverse-shield]][microverse-url]
<div>
<h1 align="center"> ⚠️ 🛠️ ⚠️ 🛠️ Changes Required 🛠️ ⚠️ 🛠️ ⚠️ </h1>
<br />
<p><b>"Fail fast, fail often"</b> and don't let this discourage you, because <b>only by failing do we learn to succeed</b>.</p>
<details open="open">
  <summary>Review Summary</summary>
  <ol>
    <li><a href="#congrats"> 💯 Awesome work 👏🏼 👏🏼 </a></li>
    <li><a href="#mandatory"> ⚠️ Things to Improve </a> 🚨 🚨 </li>
    <li><a href="#optional"> 💡Nice to have and other comments ✍🏼 ✍🏼 </a></li>
    <li><a href="#contact"> 🧔🏼 About your Reviewer </a> (you don't know it... but we are just automated bots 🤖) </li>
    <li><a href="#help"> 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼 </a></li>
  </ol>
</details>
</div>

<h3 id="congrats"> 1. :100: Awesome work</h3>
<details close="close">
<summary> [click to open] Big or little achievements, but achievements at the end. <b>Feel proud that you're learning and improving everyday.</b></summary>



</details>

<hr>
<h3 id="mandatory"> 2. ⚠️ Things to Improve (or the To-Do list to get the project approved) </h3> 
<details open="open">
<summary> [click to open] Working on these observations will help you advance and get closer to the goal 🎯, postpone them or do them half-ways and they'll haunt you 👻 👻... FOREVER!! (It will feel like that, trust me). </summary>


- [ ] enumerables `#my_none?` when no block or argument is given returns true only if none of the collection members is true



</details>
<p><b>NOTE</b>This list is or could be divided in what is show in this comment and inline comments posted below.<p>
<hr>

<h3 id="optional"> 3. 💡 Nice to have and other comments </h3>
<details close="close">
<summary> [click to open] </h3></summary>


- [x] Nothing more to add.


</details>

<h3 id="contact"> 4. 🧔🏼 About your Reviewer </h3>
<details close="close">
<summary> [click to open] Say "hello world!" on Slack: @jubaan</summary>
<br />
<div align="center">
  <a src="https://www.github.com/jubaan">
    Julio Añoveros
  </a>
  <p>
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    🤖 Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
</div>
</details>

<h3 id="help"> 5. 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼</h3>
<details close="close">
  <summary> [click to open] <b>Feel stuck?</b> </summary>
  <br>
  <p> First breath... just do it. Take 15-20 min away from the computer, stretch, meditate, do some exercise, drink water, or do whatever is necessary to distract your mind from the obstacle.</p>
  <p><b>Sometimes you just need to get away to see things with a new perspective.</b></p>
</details>

<details close="close">
  <summary> [click to open] <b>Need help?</b> </summary>
  <br>
  <p> Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack, you'll find me as @jubaan. If I'm not available there are other Code Reviewers you can contact. They'll help as well.</p>
  <p>Remember that <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> has other channels where you can get an answer to your questions:</p>
  <ol>
    <li>The <b>students channel 🧑‍🎓 </b> in slack,</li>
    <li>your <b>teammates 👥 </b>,</li>
    <li>your <b>mentor 👨🏼 👩🏼 </b>, and</li>
    <li><b>Student Success 🏫 </b>.</li>
  </ol>
  <p><b> DON'T hesitate to ask! </b></p>
</details>
<p>If all of the above doesn't help... get a <a href="https://rubberduckdebugging.com/">rubber duck</a> ... it would be great to own one 😄. </p>

<h2 align="center"> Keep up the great work :muscle: and Happy Coding :smiley: :computer:!! </h1>

[microverse-shield]: https://img.shields.io/badge/-microverse-%236F23FF?style=for-the-badge
[microverse-url]: https://www.microverse.org/?grsf=kfm2ej
[bootstrap-shield]: https://img.shields.io/badge/Bootstrap-white?style=for-the-badge&logo=bootstrap
[bootstrap-url]: https://www.microverse.org/?grsf=kfm2ej

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @SamanAtashi @danmainah
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

There is just 1 test case that is failing. Kindly work on the suggested changes 👇.
- [ ] #my_none? when no block or argument is given returns true only if none of the collection members is true
```
     Failure/Error: expect(false_array.my_none?).to be true
     
       expected true
            got false
```


##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad (Code Reviewer)](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Comment by mimipeshy** [comment]
## STATUS: APPROVED 🥳 
Hello there @SamanAtashi and @danmainah 👋🏾 👋🏾 great job so far on this milestone. Most of your test cases pass very well, you have worked on the TSE changes, keep it up 😄 

**[OPTIONAL]** 🟢 
Although I have marked this project as approved and given the optional tag, it would be nice if you fixed the following error before filling in the project completion form. This will help you avoid errors when you will have to test the enumerable methods created.
- [ ]  enumerables #my_none? when no block or argument is given returns true only if none of the collection members is true
     Failure/Error: expect(false_array.my_none?).to be true
     
       expected true
            got false
#### [HINT]: The secret with #my_none? the method is that it's the opposite of the #my_any method, so where its true in#my_any should return a false in #my_none. So in your case, it should look like this ⬇️ 
```
def my_none?(sth = nil)
    new_self = *self
    return true if new_self.empty?

    if block_given?
      new_self.my_each do |item|
        return false if yield item
      end
    else
      new_self.my_each do |item|
        if sth.instance_of?(Regexp)
          return false if item.match?(sth)
        elsif sth.is_a? Class
          return false if item.is_a? sth
        elsif sth.nil?
          to_a.my_each { |i| return false if i }
        elsif [true].include?(new_self)
          return false
        elsif sth == item
          return false
        end
      end
    end
    true
  end
```
Once you have fixed the above error, go ahead and merge. All the best 😄 

Regards
Peris Ndanu

**Pull request merged**